engine/src/main/java/org/camunda/bpm/engine/impl/interceptor/ProcessDataContext.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.interceptor;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.camunda.bpm.application.ProcessApplicationReference;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.camunda.bpm.engine.impl.context.Context;
+import org.camunda.bpm.engine.impl.persistence.entity.ExecutionEntity;
+import org.camunda.commons.logging.MdcAccess;
+
+/**
+ * Holds the contextual process data.<br>
+ *
+ * New context properties are always part of a section that can be started by
+ * {@link #pushSection(ExecutionEntity)}. The section keeps track of all pushed
+ * properties. Those can easily be cleared by popping the section with
+ * {@link #popSection()} afterwards, e.g. after the successful execution.<br>
+ *
+ * A property can be pushed to the logging context (MDC) if there is a configured
+ * non-empty context name for it in the {@link ProcessEngineConfigurationImpl
+ * process engine configuration}. The following configuration options are
+ * available:
+ * <ul>
+ * <li>loggingContextActivityId - the context property for the activity id</li>
+ * <li>loggingContextApplicationName - the context property for the application name</li>
+ * <li>loggingContextBusinessKey - the context property for the business key</li>
+ * <li>loggingContextDefinitionId - the context property for the definition id</li>
+ * <li>loggingContextProcessInstanceId - the context property for the instance id</li>
+ * <li>loggingContextTenantId - the context property for the tenant id</li>
+ * </ul>
+ */
+public class ProcessDataContext {
+
+  private static final String NULL_VALUE = "~NULL_VALUE~";
+
+  public static final String PROPERTY_ACTIVITY_ID = "activityId";
+
+  private String mdcPropertyActivityId;
```

Comment:
Since we're refactoring, should the fields be set to `protected` as per the code style?

Reply:
Absolutely! 👍